### PR TITLE
Force parentheses around `match` expression in binary expression

### PR DESCRIPTION
**Pull request by bwmf2**
This attempts to solve https://github.com/rust-lang/rust/issues/98790.

### CODE REVIEW

**Comment by compiler-errors** [comment]
This needs a test

**Comment by bwmf2** [comment]
I added the test.

File: compiler/rustc_ast_pretty/src/pprust/state/expr.rs [hunk 1]
```diff
@@ -244,6 +244,7 @@ impl<'a> State<'a> {
             (&ast::ExprKind::Let { .. }, _) if !parser::needs_par_as_let_scrutinee(prec) => {
                 parser::PREC_FORCE_PAREN
             }
+            (&ast::ExprKind::Match(..), _) => parser::PREC_FORCE_PAREN,
```
Comment (wesleywiser) [inline compiler/rustc_ast_pretty/src/pprust/state/expr.rs:247]: Can you add a comment here like the other two cases which provides an example of why this is needed?

**Review by wesleywiser — COMMENTED** [review]
Looks good to me with the suggested comment added!

File: compiler/rustc_ast_pretty/src/pprust/state/expr.rs (hunk 1 above)
Comment (bwmf2) [reply compiler/rustc_ast_pretty/src/pprust/state/expr.rs:250]: Added.

**Pull request merged**